lib/iex/lib/iex/introspection.ex

```diff
@@ -644,7 +644,16 @@ defmodule IEx.Introspection do
         types_not_found(inspect(module))
 
       {:ok, types} ->
-        Enum.each(types, &(&1 |> format_type() |> IO.puts()))
+        types
+        |> Enum.filter(&match?({kind, _} when kind in [:type, :opaque], &1))
```

Comment:
@josevalim I also filtered out `typep` here, which was not done before.

Reply:
According to #2612 we want to list typeps so Id be careful about making this particular change.

Reply:
I think we should add a test to prevent `typep` to be filtered out in future PRs

Reply:
@wojtekmach That's why I'm specifically pointing it out. Should I leave it in or should I remove the line?

Before this PR we've shown `typep` for `t(module)`, but not for `t({module, type})` / `t({module, type, arity})`. This didn't make sense to me, therefore I changed the behaviour of the first to reflect the others.

Reply:
@fertapric If the behavior from before is the desired one, I'll remove the line and can also add a test to make sure it is not filtered out in the future.

Reply:
@fertapric & @wojtekmach Adressed in 841f5ab

Reply:
@maennchen @fertapric thanks for adding the test <3